pkg/controllers/ocmcontroller/ocmcontroller.go

```diff
@@ -112,189 +99,64 @@ func runControllers(ctx context.Context,
-			csrReconciles,
-			controllerContext.EventRecorder,
-		)
-	}
-
-	leaseController := lease.NewClusterLeaseController(
-		kubeClient,
-		clusterClient,
-		clusterInformers.Cluster().V1().ManagedClusters(),
-		kubeInformers.Coordination().V1().Leases(),
-		controllerContext.EventRecorder,
-	)
-
-	rbacFinalizerController := rbacfinalizerdeletion.NewFinalizeController(
-		kubeInformers.Rbac().V1().Roles(),
-		kubeInformers.Rbac().V1().RoleBindings(),
-		kubeInformers.Core().V1().Namespaces().Lister(),
-		clusterInformers.Cluster().V1().ManagedClusters().Lister(),
-		workInformers.Work().V1().ManifestWorks().Lister(),
-		kubeClient.RbacV1(),
-		controllerContext.EventRecorder,
-	)
-
-	managedClusterSetController := managedclusterset.NewManagedClusterSetController(
-		clusterClient,
-		clusterInformers.Cluster().V1().ManagedClusters(),
-		clusterInformers.Cluster().V1beta2().ManagedClusterSets(),
-		controllerContext.EventRecorder,
-	)
-
-	managedClusterSetBindingController := managedclustersetbinding.NewManagedClusterSetBindingController(
-		clusterClient,
-		clusterInformers.Cluster().V1beta2().ManagedClusterSets(),
-		clusterInformers.Cluster().V1beta2().ManagedClusterSetBindings(),
-		controllerContext.EventRecorder,
-	)
-
-	clusterroleController := clusterrole.NewManagedClusterClusterroleController(
-		kubeClient,
-		clusterInformers.Cluster().V1().ManagedClusters(),
-		kubeInformers.Rbac().V1().ClusterRoles(),
-		controllerContext.EventRecorder,
-	)
-
-	addOnHealthCheckController := addon.NewManagedClusterAddOnHealthCheckController(
 		addOnClient,
-		addOnInformers.Addon().V1alpha1().ManagedClusterAddOns(),
-		clusterInformers.Cluster().V1().ManagedClusters(),
-		controllerContext.EventRecorder,
+		kubeInformers,
+		clusterInformers,
+		workInformers,
+		addOnInformers,
 	)
 
-	addOnFeatureDiscoveryController := addon.NewAddOnFeatureDiscoveryController(
+	go placementcontrollers.RunControllerManagerWithInformers(
+		ctx,
+		controllerContext,
+		kubeClient,
 		clusterClient,
-		clusterInformers.Cluster().V1().ManagedClusters(),
-		addOnInformers.Addon().V1alpha1().ManagedClusterAddOns(),
-		controllerContext.EventRecorder,
+		clusterInformers,
 	)
 
-	var defaultManagedClusterSetController, globalManagedClusterSetController factory.Controller
-	if features.DefaultControlplaneMutableFeatureGate.Enabled(ocmfeature.DefaultClusterSet) {
-		defaultManagedClusterSetController = managedclusterset.NewDefaultManagedClusterSetController(
-			clusterClient.ClusterV1beta2(),
-			clusterInformers.Cluster().V1beta2().ManagedClusterSets(),
-			controllerContext.EventRecorder,
-		)
-		globalManagedClusterSetController = managedclusterset.NewGlobalManagedClusterSetController(
-			clusterClient.ClusterV1beta2(),
-			clusterInformers.Cluster().V1beta2().ManagedClusterSets(),
-			controllerContext.EventRecorder,
+	if features.HubMutableFeatureGate.Enabled(ocmfeature.ManifestWorkReplicaSet) {
+		go workhub.RunControllerManagerWithInformers(
```

Comment:
This also will enable the work hub controller (ManifestWorkReplicaSet) 
/cc @tamalsaha